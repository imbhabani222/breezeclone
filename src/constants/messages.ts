const messages = {
  NOT_AUTHORIZED:
    "You are not authorized or session has been expired. Please try again.",
  NO_DATA_AVAILABLE: "No data available.",
  ACCESS_DENIED: "Access Denied/Forbidden.",
  DEFAULT_ERROR_MESSAGE:
    "There seems to be a problem, please retry after sometime.",
  USER_NOT_FOUND: "Invalid user",
  IMAGE_NOT_AVAILABLE: "Image not available",
  ARE_YOU_SURE: "Are you sure?",
  TYPE_OUTLINE: "outline",
  TYPE_RECTFILLED: "rectFilled",
  TYPE_FILLED: "filled",
  PLEASE_TYPE_ANSWER: "Please type your answer.",
  EXAM_NAME: "[0-2 years] Data Analytics",
  EXAM_TYPE: "By DoSelect UI/UX Testing",
  MODAL_CONTENT:
    "# Are you sure ? \n You still have **%hours% hours %minutes% mins %seconds% seconds** left to submit this assessment. </br></br>You have attempted %completed% of %total% questions.</br></br>You will not be able to attempt any question or make changes to your solution once you submit.",
  LOADING: "Loading...",
  PROCEED_TO_FEEDBACK: "PROCEED TO FEEDBACK PAGE",
  GOING_GREAT:
    "Going great .</br></br>You have attempted %completed% of %total% questions.",
  AUTO_NEXT_MODAL:
    "This section has been auto-submitted as the allotted time is up.</br></br>You have attempted %completed% of %total% questions.",
  BROWSER_TOLERANCE_INFO:
    "  You had navigated away from the test screen. This incident will be reported to the test administrator.",
  LOAD_MORE: "Load More",
  BACKBUTTON: " Go back to Assessment",
  PLATFORM_SUPPORT: "Platform Support",
  CLIENTS_SUPPORT: "Clients Support",
  HELP: "Help",
  QUESTION: "View All Questions",
  RIGHT: "right",
  FEEDBACK: {
    FEEDBACKCOMPLETED: "Woot! You've completed this assessment.",
    FEEDBACKTITLE: "SHARE FEEDBACK",
    FEEDBACKNOTIFIED:
      "We've notified team %orgName% about this. If you have any queries regarding this, please get in touch with your point of contact.",
    SENDFEEDBACK: "Send Feedback",
    ENTERFEWWORDS: "Please enter few words",
    MESSAGE_LABEL: "message",
    SUBMIT_LABEL: "Submit",
    FEEDBACKFORM: "feedbackForm",
    LIKETOTELL: "Is there something that you'd like to tell us?",
  },
  REPORT_LABLE: "REPORT AN ISSUE",
  REPORT_MODAL_TITLE: "Report an issue with Stack Using Interface",
  REPORT_MODAL_INPUT_LABEL: "  Other:",
  PROGRESS_DATABAR_TEXT: "%completed%/%total% Questions attempted",
  RUN_CODE_LABEL: "Run Code",
  ABORT_LABEL: "Abort",
  TEST_RESULTS_LABEL: "Test results",
  TEST_RESULTS_TEXT: 'Click on "Run code" to evaluate your solution',
  CUSTOM_INPUT_LABEL: "Custom Input",
  CUSTOM_INPUT_TEXT: "STDIN",
  ICON: "Icon",
  TIME_LABEL: {
    H: "H:",
    M: "M:",
    S: "S",
  },
  FAQ: {
    KEY1: "1",
    KEY2: "2",
    KEY3: "3",
    TAB1: "Help",
    TAB2: "Coding Environment",
    TAB3: "Status Code",
    SEARCH_PLACEHOLDER: "Ask a question",
    FAQINFOTEXT: `Please find below the general instructions and information about 
    DoSelect's assessment environment. We highly recommend that you go
    through this before you start taking the assessment.`,
  },
  TESTSTATUS: {
    ICONSTATUS: "iconsStatus",
    ACTIVATEICON: "activateIcon",
    DEACTIVATEICON: "deactivateIcon",
  },
  DESCRIPTION_LABEL: "Description",
  STATUS_CODE_LABEL: "Status Code",
  LIBRARIES_AVAILABLE_LABEL: "Libraries Available",
  TIME_LIMIT_LABEL: "Time Limit(in seconds)",
  MEMORY_UNIT_LABEL: "Memory Limit (in MB)",

  LIBRARIES_AVAILABLE_KEY: "Libraries available",
  TIME_LIMIT_KEY: "Time limit (in seconds)",
  MEMORY_UNIT_KEY: "Memory limit (in MB)",

  VERSION_LABEL: "Version",
  LANGUAGE_LABEL: "Language",
  DO_SELECT_RESEVE_RIGHTS: "Doselect. All rights reserved",
  LOGO_LABEL: "Logo",
  TEXT_LABEL: "text",
  TIME_FOR_EXAM_LABEL: "%hours% hours %minutes% mins",
  TIME_FOR_MODAL: "%hours% hours %minutes% mins %seconds% seconds",
  ALT_IMG: "icon Doselect",
  CANCEL: "Cancel",
  SUBMIT_END: "Submit and end the Assessment",
  SUBMIT_MOVE_NEXT: "Submit and Move to Next Section",
  PROCEED_TO_NEXT: "Proceed to next section",
  UPLOAD: "Upload",
  SAVE: "Save",
  QUESTION_TYPE: "MCQ + Coding + Fill in the  Blanks",
  LOGO: "do select logo",
  CAMERA: "Video Proctoring",
  NETWORK: "Network quality",
  TIMER: "Timer",

  BROWSER_FULLSCREEN: `<p>This %type% requires your browser window to be in fullscreen mode</p>`,
  UPLOAD_ATTACHMENT: "Upload an attachment with this solution.",
  VIDEO_TEST: {
    DESC: "Description",
    VID_DUR: "Video Duration",
    SEC: "seconds",
    RECORD: "Record",
    STOP: "Stop",
    RESUME: "Resume",
    PAUSE: "Pause",
    SUBMIT: "Submit",
    REC_DUR: "Record duration:",
  },

  CODING: {
    CODING: "Coding",
    RELOAD_STUB: "Reload Code Stub?",
    RELOAD_STUB_CONTENT: `You have changed the technology of this solution. Do you want
    to REPLACE your current code with the default code stub of
    this technology? Please note that you will lose your current
    code by doing this.`,
    NO_TEST_FOUND: "No Test Cases Found",
    RUN_ALL_TEST_CASES: "Run all Test Cases",
    DESCRIPTION: "Description",
    DESCRIPTIONS: "Descriptions",
    INPUT: "Input",
    OUTPUT: "Output",
    EXPECTED_OUTPUT: "Expected Output",
    EVAL_TIME: "Eval Time",
    CPU: "CPU",
    MEMORY: "Memory",
    EXECUTION_LOGS: "Execution Logs",
    ANNOTATION: "Annotation",
  },
};

export default messages;
