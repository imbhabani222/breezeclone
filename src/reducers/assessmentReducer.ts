import actionTypes from "../constants/actionTypes/assessmentActionsTypes";

import {
  assessmentDetails,
  assessmentState,
  IdentityActions,
  TestDetailsActions,
  IdentityState,
  TechnologyActions,
} from "../types/types";

const initialIdentity: IdentityState = {
  pending: false,
  identity: [],
  error: null,
  timeLeft: Date.now() + 904444,
  declarationStatus: false,
  modalStatus: false,
};
const initialState: assessmentState = {
  loading: false,
  details: {},
  error: null,
};
export const getIdentity = (
  state = initialIdentity,
  action: IdentityActions
) => {
  switch (action.type) {
    case actionTypes.FETCH_IDENTITY:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.FETCH_IDENTITY_SUCCESS:
      return {
        ...state,
        loading: false,
        identity: action.payload.identity,
        error: null,
      };
    case actionTypes.FETCH_IDENTITY_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const getTestDetails = (
  state = initialState,
  action: TestDetailsActions
) => {
  switch (action.type) {
    case actionTypes.TEST_DETAILS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.TEST_DETAILS_SUCCESS:
      return {
        ...state,
        loading: false,
        details: action.payload.details,
      };
    case actionTypes.TEST_DETAILS_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const getTechnologyDetails = (
  state = initialState,
  action: TechnologyActions
) => {
  switch (action.type) {
    case actionTypes.TECHNOLOGY_DETAILS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.TECHNOLOGY_DETAILS_SUCCESS:
      return {
        ...state,
        loading: false,
        details: action.payload.details,
      };
    case actionTypes.TECHNOLOGY_DETAILS_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};
