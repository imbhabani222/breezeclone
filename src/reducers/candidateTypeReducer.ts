import { combineReducers } from "redux";

import actionTypes from "../constants/actionTypes/typeHackerRecruiter";
import { CandidateActions } from "../types/candidateTypes";

const hackerInitialState: any = {
  loading: false,
  data: {},
  error: null,
};
const recruiterInitialState: any = {
  loading: false,
  data: {},
  error: null,
};
export const getHackerDetails = (
  state = hackerInitialState,
  action: CandidateActions
) => {
  switch (action.type) {
    case actionTypes.FETCH_HACKER:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.FETCH_HACKER_SUCCESS:
      return {
        ...state,
        loading: false,
        data: action.payload.data,
      };
    case actionTypes.FETCH_HACKER_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const getRecruiterDetails = (
  state = recruiterInitialState,
  action: CandidateActions
) => {
  switch (action.type) {
    case actionTypes.FETCH_RECRUITER:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.FETCH_RECRUITER_SUCCESS:
      return {
        ...state,
        loading: false,
        data: action.payload.data,
      };
    case actionTypes.FETCH_RECRUITER_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};
export const getCandidateSolSet = (
  state = recruiterInitialState,
  action: CandidateActions
) => {
  switch (action.type) {
    case actionTypes.FETCH_TEST_SOLUTION:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.FETCH_TEST_SOLUTION_SUCCESS:
      return {
        ...state,
        loading: false,
        data: action.payload.data,
      };
    case actionTypes.FETCH_TEST_SOLUTION_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const getUserType = (
  state = recruiterInitialState,
  action: CandidateActions
) => {
  switch (action.type) {
    case actionTypes.FETCH_USER_TYPE:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.FETCH_USER_SUCCESS:
      return {
        ...state,
        loading: false,
        data: action.payload.data,
      };
    case actionTypes.FETCH_USER_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};
export const combineReducer = combineReducers({
  getRecruiterDetails,
  getHackerDetails,
  getCandidateSolSet,
  getUserType,
});
