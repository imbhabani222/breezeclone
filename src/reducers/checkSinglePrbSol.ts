import actionTypes from "../constants/actionTypes";
import { QuestionsAction, SinglePrbSol } from "../types/questionsActionTypes";

const initialStateProblem: any = {
  pending: false,
  data: "",
  error: null,
  statusCode: null,
};

// eslint-disable-next-line import/no-anonymous-default-export
export default (state = initialStateProblem, action: SinglePrbSol) => {
  switch (action.type) {
    case actionTypes.FETCH_SINGLE_PRB_SOL_REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.FETCH_SINGLE_PRB_SOL_SUCCESS:
      return {
        ...state,
        pending: false,
        data: action.payload,
        error: null,
        statusCode: action.payload.statusCode,
      };
    case actionTypes.FETCH_SINGLE_MCQ_SOL_FAILURE:
      return {
        ...state,
        pending: false,
        singleProblem: "",
        error: action.payload.error,
        statusCode: action.payload.statusCode.status,
      };
    default:
      return {
        ...state,
      };
  }
};
