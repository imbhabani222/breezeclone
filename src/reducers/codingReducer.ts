import { combineReducers } from "redux";
import actionTypes from "../constants/actionTypes";
import {
  LanguageAction,
  RunCodeCustomAction,
  PostSolutionsSubmitAction,
  RunCodeCustomResult,
} from "../types/codingActionTypes";

const initialState = {
  loading: false,
  data: {},
  error: null,
};
// eslint-disable-next-line import/no-anonymous-default-export

export const languageReducer = (
  state = initialState,
  action: LanguageAction
) => {
  switch (action.type) {
    case actionTypes.GET_LANGUAGE_REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.GET_LANGUAGE_SUCCESS:
      return {
        ...state,
        pending: false,
        data: action.payload.data,
        error: null,
      };
    case actionTypes.GET_LANGUAGE_FAILURE:
      return {
        ...state,
        pending: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const runCodeCustomReducer = (
  state = initialState,
  action: RunCodeCustomAction
) => {
  switch (action.type) {
    case actionTypes.RUN_CODE_CUSTOM_INPUT_REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.RUN_CODE_CUSTOM_INPUT_SUCCESS:
      return {
        ...state,
        pending: false,
        data: action.payload.data,
        error: null,
      };
    case actionTypes.RUN_CODE_CUSTOM_INPUT_FAILURE:
      return {
        ...state,
        pending: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const postSolutionsSubmitReducer = (
  state = initialState,
  action: PostSolutionsSubmitAction
) => {
  switch (action.type) {
    case actionTypes.TESTCASES_RUN_CODE__REQUEST:
      return {
        ...state,
        pending: true,
      };
    case actionTypes.TESTCASES_RUN_CODE__SUCCESS:
      return {
        ...state,
        pending: false,
        data: action.payload.data,
        error: null,
      };
    case actionTypes.TESTCASES_RUN_CODE__FAILURE:
      return {
        ...state,
        pending: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};
