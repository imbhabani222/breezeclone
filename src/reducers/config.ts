export const defaultState = {
  fetching: false,
  result: null,
  error: null,
};

export const actionsDefaultState = {
  showModal: false,
  declarationStatus: false,
  timeLeft: 0,
  checkStatus: false,
  modalStatus: false,
  timerStatus: false,
  systemCheckStatus: false,
  getSocketArgs: [],
  getVidSignedUrl: [],
  getCodeAsyncData: [],
  getCodeEvaluatedData: [],
};
