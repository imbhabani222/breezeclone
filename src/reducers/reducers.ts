import actionTypes from "../constants/actionTypes";

import { TodoActions, TodoState } from "../types/types";

const initialState: TodoState = {
  pending: false,
  todos: [],
  error: null
};

// eslint-disable-next-line import/no-anonymous-default-export
export default (state = initialState, action: TodoActions) => {
  switch (action.type) {
    case actionTypes.FETCH_TODO_REQUEST:
      return {
        ...state,
        pending: true
      };
    case actionTypes.FETCH_TODO_SUCCESS:
      return {
        ...state,
        pending: false,
        todos: action.payload.todos,
        error: null
      };
    case actionTypes.FETCH_TODO_FAILURE:
      return {
        ...state,
        pending: false,
        todos: [],
        error: action.payload.error
      };
    default:
      return {
        ...state
      };
  }
};
