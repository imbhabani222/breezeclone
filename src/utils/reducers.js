/* eslint-disable no-prototype-builtins */
export const createReducer = (initialState, handlers) => {
  return function reducer(state = initialState, action) {
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action);
    } else {
      return state;
    }
  };
};

export const getFetchingState = (state) => ({
  ...state,
  fetching: true,
  result: null,
  error: null,
});

export const getSuccessState = (state, { result }) => ({
  ...state,
  fetching: false,
  result: result.result || result,
  error: null,
});

export const getErrorState = (state, { error }) => ({
  ...state,
  fetching: false,
  result: null,
  error,
});

export const getResetState = (state) => ({
  ...state,
  fetching: false,
  result: null,
  error: null,
});

export const getActionInitialState = (state) => ({
  ...state,
});

export const getActionDefaultState = (state, { result: { data } }) => ({
  ...state,
  ...data,
});

export const checkAction = (state = {}, action) => {
  switch (action.type) {
    case "CHECK_ACTION": {
      console.log(state, action);
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};

export const customResult = (state = {}, action) => {
  switch (action.type) {
    case "GET_CUSTOM_RESULT": {
      console.log(state, action, "custommmmmmmmmm");
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};

export const testCasesResult = (state = {}, action) => {
  switch (action.type) {
    case "GET_TESTCASES_RESULT": {
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};
