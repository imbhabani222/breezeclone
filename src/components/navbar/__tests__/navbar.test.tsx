import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar, { NavbarProps } from "../navbar";
import messages from "../../../constants/messages";
import { combineReducers, createStore } from "redux";
import { getTestDetails } from "../../../reducers/assessmentReducer";
import { Provider } from "react-redux";
import VideoProctor from "../videoProctor/videoProctor";

const { HELP, LOGO, CAMERA, NETWORK, QUESTION, TIMER } = messages;

describe("Navbar", () => {
  const initialState = {
    loading: false,
    details: {},
    error: null,
  };

  const mocks = { timeLeft: 1645761486931 };

  const fakeStore = createStore(
    combineReducers({ testDetails: getTestDetails }),
    {
      testDetails: initialState,
    }
  );

  fakeStore.dispatch = jest.fn((): any => {
    return mocks;
  });
  // const ActualProctor = jest.requireActual("../videoProctor/videoProctor");
  // jest.mock("../videoProctor/videoProctor", () => (
  //   <div data-testid="video-proctor">
  //     <ActualProctor />
  //   </div>
  // ));

  const defaultProps: NavbarProps = {
    network: false,
    // que_label: false,
    // video_camera: false,
    // clock: false,
  };

  function renderNavbar(props: Partial<NavbarProps> = {}) {
    return render(
      <Provider store={fakeStore}>
        <BrowserRouter>
          <Navbar {...defaultProps} {...props} />
        </BrowserRouter>
      </Provider>
    );
  }

  afterEach(cleanup);
  test("snapshot matches with Navbar component", () => {
    const { asFragment } = renderNavbar();
    expect(asFragment()).toMatchSnapshot();
  });

  test("should display a plain navbar with no icons", async () => {
    renderNavbar();
    expect(screen.getByText(HELP)).toBeInTheDocument();
  });
  test("logo display test", async () => {
    renderNavbar();
    expect(screen.getByTitle(LOGO)).toBeInTheDocument();
  });
  test("do not display camera icon", async () => {
    renderNavbar();
    expect(screen.queryByTitle(CAMERA)).not.toBeInTheDocument();
  });
  test("display camera icon", async () => {
    renderNavbar({ video_camera: true });
    // expect(screen.getByTestId(/video-proctor/)).toBeInTheDocument();
    expect(screen.getByTitle(CAMERA)).toBeInTheDocument();
  });
  test("click on logo icon", async () => {
    renderNavbar();
    fireEvent(
      screen.getByTitle(LOGO),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    );
  });
  test("do not display network icon", async () => {
    renderNavbar();
    expect(screen.queryByTitle(NETWORK)).not.toBeInTheDocument();
  });
  test("display network icon", async () => {
    renderNavbar({ network: true });
    expect(screen.getByTitle(NETWORK)).toBeInTheDocument();
  });
  //display help button
  test("display help button", async () => {
    renderNavbar();
    expect(screen.getByText(HELP)).toBeInTheDocument();
  });
  //click help button
  test("click on help icon", async () => {
    renderNavbar();
    fireEvent(
      screen.getByTitle(HELP),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    );
  });
  // test("do not display all questions button", async () => {
  //   renderNavbar();
  //   expect(screen.queryByTitle(QUESTION)).not.toBeInTheDocument();
  // });
  //display all question
  test("display all questions button", async () => {
    renderNavbar();
    expect(screen.getByTitle(QUESTION)).toBeInTheDocument();
  });
  //click all question
  test("click on all questions icon", async () => {
    renderNavbar();
    fireEvent(
      screen.getByTitle(QUESTION),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    );
  });
  test("more content popover is not visible", () => {
    renderNavbar();
    const nullPopover = screen.queryByTestId("more-popover");
    expect(nullPopover).toBeInTheDocument();
  });
  test("more content popover is visible", async () => {
    renderNavbar();
    const handleClick = jest.fn();
    fireEvent.click(screen.getByTestId("more-popover"));
    // expect(handleClick).toHaveBeenCalledTimes(1);
    expect(screen.getByTestId("more-popover")).toHaveBeenCalledTimes(1);
    expect(await screen.findByTestId("more-popover")).toBeInTheDocument();
  });
  // test("do not display timer", async () => {
  //   renderNavbar();
  //   expect(screen.queryByTitle(TIMER)).not.toBeInTheDocument();
  // });
  // test("display timer", async () => {
  //   renderNavbar({ clock: true });
  //   expect(screen.getByTitle(TIMER)).toBeInTheDocument();
  // });
});
