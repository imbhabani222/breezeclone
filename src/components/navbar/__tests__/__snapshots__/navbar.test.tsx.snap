// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot matches with Navbar component 1`] = `
<DocumentFragment>
  <header
    class="ant-layout-header site_page_header"
  >
    <span
      class="anticon logo"
      role="img"
      tabindex="-1"
      title="do select logo"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        focusable="false"
        height="1em"
        width="1em"
      >
        logo.svg
      </svg>
    </span>
    <div
      class="que_label"
      title="View All Questions"
    >
      View All Questions
    </div>
    <div
      class="content_div"
    >
      <div
        class="helpiconbtn"
      >
        <span
          class="anticon icon"
          role="img"
          title="Help"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            help.svg
          </svg>
        </span>
        <div
          class="ant-typography text"
        >
          Help
        </div>
      </div>
      <div>
        <span
          aria-label="more"
          class="anticon anticon-more view_more_content"
          data-testid="more-popover"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="more"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
